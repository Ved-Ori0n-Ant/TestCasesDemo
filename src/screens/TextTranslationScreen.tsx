import React, {useState} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  Text,
  StatusBar,
  ScrollView,
  Pressable,
} from 'react-native';
import '../localization';
import {useTranslation} from 'react-i18next';
import LanguageSelector from '../utils/languageSelector';
import Section from '../utils/section';
import NextnPrevious, { Stack } from '../components/NextnPrevious';

declare const global: {HermesInternal: null | {}};

const TextTranslationScreen = () => {
  const {t} = useTranslation();

  const user = {
    username: 'Rajesh',
  };

  const userBill = {
    total: 420.69,
    paymentDueDate: 'May the fourth',
  };

  const genderPronouns = ['she', 'he', 'they', 'ze', 'ey'];
  const randomMessages: any = t('randomLoadingMessages', {returnObjects: true});
  return (
    <>
      <StatusBar barStyle="light-content" backgroundColor="black" />
      <SafeAreaView style={styles.main}>
        <ScrollView style={styles.scroll} showsVerticalScrollIndicator={false}>
          <Section testID='NamespacesSection' title="NAMESPACES">
            <Text>{t('homePage:welcome')}</Text>
            <Text>{t('ok')}</Text>
          </Section>
          <Section  testID='ChangeLanguageSection' title="CHANGE LANGUAGE">
            <LanguageSelector />
          </Section>
          <Section testID='InterpolationSection' title="INTERPOLATION">
            <Text>{t('accountSuspended', {user, userBill})}</Text>
          </Section>
          <Section testID='PluralizationSection' title="PLURALIZATION">
            <Text>{t('autopsy', {count: 0})}</Text>
            <Text>{t('autopsy', {count: 1})}</Text>
            <Text>{t('autopsy', {count: 7})}</Text>
          </Section>
          <Section testID='ContextGenderSection' title="CONTEXT: GENDER">
            {genderPronouns.map(context => (
              <Text key={context}>{t('loveThyself', {context})}</Text>
            ))}
          </Section>
          <Section title="CONTEXT: PLURALIZATION">
            <Text>{t('autopsy', {count: 0, context: '0'})}</Text>
            <Text>{t('autopsy', {count: 1, context: '1'})}</Text>
            <Text>{t('autopsy', {count: 7, context: '7'})}</Text>
          </Section>
        </ScrollView>
      </SafeAreaView>
      <NextnPrevious testID='np_btn' currentScreen={Stack.TextTranslationScreen} />
    </>
  );
};

const styles = StyleSheet.create({
  main: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    // width: '100%',
  },
  scroll: {
    flex: 1,
  },
});

export default TextTranslationScreen;
