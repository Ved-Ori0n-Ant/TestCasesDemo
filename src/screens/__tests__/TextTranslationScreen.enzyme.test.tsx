import React from 'react';
import TextTranslationScreen from '../TextTranslationScreen';
import {shallow} from 'enzyme';

const mockedNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    dispatch: mockedNavigate,
  }),
}));

jest.mock('@react-native-community/async-storage', () => ({
  getItem: jest.fn(),
}));

jest.mock('react-native-localize', () => {});

const jumbleJumble = jest.fn();

const useTranslation = jest.fn();

describe('tests for TextTranslationScreen', () => {
  it('should render okay', () => {
    const tree = shallow(<TextTranslationScreen />);
    expect(tree).toMatchSnapshot();
  });
  it('should have namespaces', () => {
    const tree = shallow(<TextTranslationScreen />);
    const nsSection = tree.findWhere(
      node => node.prop('testID') === 'NamespacesSection',
    );
    expect(nsSection).not.toBeNull();
    expect(nsSection).toBeDefined();
    expect(nsSection.props().title).toBe('NAMESPACES');
    expect(nsSection.props().children[0].props.children).toBe('welcome');
    expect(nsSection.props().children[1].props.children).toBe('ok');
  });
  it('should have all sections', () => {
    const tree = shallow(<TextTranslationScreen />);
    const lngChangeSection = tree.findWhere(
      node => node.prop('testID') === 'ChangeLanguageSection',
    );
    const interpolationSection = tree.findWhere(
      node => node.prop('testID') === 'InterpolationSection',
    );
    const pluralizationSection = tree.findWhere(
      node => node.prop('testID') === 'PluralizationSection',
    );
    const genderSection = tree.findWhere(
      node => node.prop('testID') === 'ContextGenderSection',
    );

    expect(lngChangeSection.props().title).toBe('CHANGE LANGUAGE');
    expect(interpolationSection.props().title).toBe('INTERPOLATION');
    expect(pluralizationSection.props().title).toBe('PLURALIZATION');
    expect(genderSection.props().title).toBe('CONTEXT: GENDER');
    expect(interpolationSection.props().children.props.children).toBe(
      'accountSuspended',
    );
    expect(pluralizationSection.props().children[0].props.children).toBe(
      'autopsy',
    );
    expect(genderSection.props().children[0].props.children).toBe(
      'loveThyself',
    );
  });
  it('should render next n prev buttons', () => {
    const tree = shallow(<TextTranslationScreen />);
    const npBtn = tree.findWhere(node => node.prop('testID') === 'np_btn');
    const prevbtn = npBtn
      .dive()
      .findWhere(node => node.props().testID === 'Previous_btn');
    prevbtn.props().onPress();
    expect(mockedNavigate).toBeCalled();
    expect(mockedNavigate).toBeCalledWith({
      type: 'REPLACE',
      payload: {
        name: 'TextScreen',
      },
    });
    const nextbtn = npBtn
      .dive()
      .findWhere(node => node.props().testID === 'Next_btn');
    nextbtn.props().onPress();
    expect(mockedNavigate).toBeCalled();
    expect(mockedNavigate).toBeCalledWith({
      type: 'REPLACE',
      payload: {
        name: 'TextInputScreen',
      },
    });
  });
});
