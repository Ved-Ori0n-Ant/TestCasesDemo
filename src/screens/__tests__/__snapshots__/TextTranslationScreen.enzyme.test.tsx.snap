// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for TextTranslationScreen should render okay 1`] = `
<Fragment>
  <StatusBar
    backgroundColor="black"
    barStyle="light-content"
  />
  <RCTSafeAreaView
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ScrollView
      showsVerticalScrollIndicator={false}
      style={
        {
          "flex": 1,
        }
      }
    >
      <Section
        testID="NamespacesSection"
        title="NAMESPACES"
      >
        <Text>
          welcome
        </Text>
        <Text>
          ok
        </Text>
      </Section>
      <Section
        testID="ChangeLanguageSection"
        title="CHANGE LANGUAGE"
      >
        <LanguageSelector />
      </Section>
      <Section
        testID="InterpolationSection"
        title="INTERPOLATION"
      >
        <Text>
          accountSuspended
        </Text>
      </Section>
      <Section
        testID="PluralizationSection"
        title="PLURALIZATION"
      >
        <Text>
          autopsy
        </Text>
        <Text>
          autopsy
        </Text>
        <Text>
          autopsy
        </Text>
      </Section>
      <Section
        testID="ContextGenderSection"
        title="CONTEXT: GENDER"
      >
        <Text
          key="she"
        >
          loveThyself
        </Text>
        <Text
          key="he"
        >
          loveThyself
        </Text>
        <Text
          key="they"
        >
          loveThyself
        </Text>
        <Text
          key="ze"
        >
          loveThyself
        </Text>
        <Text
          key="ey"
        >
          loveThyself
        </Text>
      </Section>
      <Section
        title="CONTEXT: PLURALIZATION"
      >
        <Text>
          autopsy
        </Text>
        <Text>
          autopsy
        </Text>
        <Text>
          autopsy
        </Text>
      </Section>
    </ScrollView>
  </RCTSafeAreaView>
  <NextnPrevious
    currentScreen={1}
    testID="np_btn"
  />
</Fragment>
`;
